### PR TITLE
fix: Use line type and remove booking contact

### PR DESCRIPTION
**Pull request by jorelosorio**
Closes: https://github.com/AtB-AS/kundevendt/issues/4103
Depend on: https://github.com/AtB-AS/atb-bff/pull/260

Before:

image

After:

Screenshot_1685104708

### CODE REVIEW

File: src/travel-details-screens/components/Trip.tsx [hunk 1]
```diff
@@ -157,7 +157,7 @@ const useStyle = StyleSheet.createThemeHook((theme) => ({
     paddingTop: theme.spacings.medium,
   },
   container: {
-    paddingBottom: theme.spacings.medium,
+    paddingVertical: theme.spacings.medium,
```
Comment (jorelosorio) [inline src/travel-details-screens/components/Trip.tsx:160]: Before this change:

image

After the change:

image

File: src/travel-details-screens/utils.ts [hunk 2]
```diff
@@ -162,5 +162,5 @@ function someLegsAreByTrain(tripPattern: TripPattern): boolean {
 }
 
 export function isLegFlexibleTransport(leg: Leg): boolean {
-  return leg.line?.id.toLowerCase().includes('flexible') ?? false;
+  return !!leg.line?.flexibleLineType;
```
Comment (jorelosorio) [inline src/travel-details-screens/utils.ts:165]: At least not null?

File: src/travel-details-screens/components/Trip.tsx (hunk 1 above)
Comment (jorelosorio) [reply src/travel-details-screens/components/Trip.tsx]: It was fixed in another branch

**Review by gorandalum — APPROVED** [review]
Looks good! Also remember to update the `journeyplanner-types_v3.ts` as the updates in the bff should be reflected in the app.

**Comment by jorelosorio** [comment]
> Looks good! Also remember to update the `journeyplanner-types_v3.ts` as the updates in the bff should be reflected in the app.

I made a commit copying the generated file from BFF to the app, please double-check :)

File: src/travel-details-screens/utils.ts (hunk 2 above)
Comment (gorandalum) [reply src/travel-details-screens/utils.ts:165]: Seems correct!

**Pull request merged**